### ISSUE TITLE
[Bug]: decompress_chunk fails version 2.11.1

### ISSUE THREAD

**Issue by aarondglover** [timescale/timescaledb#5984]
### What type of bug is this?

Unexpected error

### What subsystems and features are affected?

Compression

### What happened?

Timescaledb v2.11.1 - decompress_chunk failing and can't update/delete from compressed chunk!

I updated to version 2.11.1 and was keen to apply compression to my hypertable.

I had not done so to this point as it was impossible (or difficult) to update and/or delete data which is a compressed chunk... but documenttaion says this is supported in version 2.11.1

```
 [autoind-mdm-34]2023-08-18T10:35:33.189902[postgresql-13][731-1] pid=730042,user=,db=,app=,client= LOG: server process (PID 858992) was terminated by signal 11: Segmentation fault
 [autoind-mdm-34]2023-08-18T10:35:33.190154[postgresql-13][731-2] pid=730042,user=,db=,app=,client= DETAIL: Failed process was running: SELECT decompress_chunk('_timescaledb_internal._hyper_14_156_chunk', true);
 [autoind-mdm-34]2023-08-18T10:35:33.190191[postgresql-13][731-3]
 [autoind-mdm-34]2023-08-18T10:35:33.190225[postgresql-13][732-1] pid=730042,user=,db=,app=,client= LOG: terminating any other active server processes
 [autoind-mdm-34]2023-08-18T10:35:33.191011[postgresql-13][733-1] pid=858471,user=postgres,db=_timescaledb,app=aiven-pruned,client=[local] WARNING: terminating connection because of crash of another server process
 [autoind-mdm-34]2023-08-18T10:35:33.191216[postgresql-13][733-2] pid=858471,user=postgres,db=_timescaledb,app=aiven-pruned,client=[local] DETAIL: The postmaster has commanded this server process to roll back the current transaction and exit, because another server process exited abnormally and possibly corrupted shared memory.

```

It simply crashes the entire server when trying to decompress_chunk on some of the chunks

and throws error

`cannot update/delete rows from chunk "_hyper_14_81_chunk" as it is compressed`

When trying to update/delete data

### TimescaleDB version affected

2.11.1

### PostgreSQL version used

13.x

### What operating system did you use?

Timescale Managed Service 

### What installation method did you use?

Other

### What platform did you run on?

Managed Service for TimescaleDB (MST/Aiven)

### Relevant log output and stack trace

```bash
[autoind-mdm-34]2023-08-18T10:35:33.189902[postgresql-13][731-1] pid=730042,user=,db=,app=,client= LOG: server process (PID 858992) was terminated by signal 11: Segmentation fault
 [autoind-mdm-34]2023-08-18T10:35:33.190154[postgresql-13][731-2] pid=730042,user=,db=,app=,client= DETAIL: Failed process was running: SELECT decompress_chunk('_timescaledb_internal._hyper_14_156_chunk', true);
 [autoind-mdm-34]2023-08-18T10:35:33.190191[postgresql-13][731-3]
 [autoind-mdm-34]2023-08-18T10:35:33.190225[postgresql-13][732-1] pid=730042,user=,db=,app=,client= LOG: terminating any other active server processes
 [autoind-mdm-34]2023-08-18T10:35:33.191011[postgresql-13][733-1] pid=858471,user=postgres,db=_timescaledb,app=aiven-pruned,client=[local] WARNING: terminating connection because of crash of another server process
 [autoind-mdm-34]2023-08-18T10:35:33.191216[postgresql-13][733-2] pid=858471,user=postgres,db=_timescaledb,app=aiven-pruned,client=[local] DETAIL: The postmaster has commanded this server process to roll back the current transaction and exit, because another server process exited abnormally and possibly corrupted shared memory.

It simply crashes the entire server when trying to decompress_chunk on some of the chunks
```

### How can we reproduce the bug?

```bash
No idea
```

**Comment by shhnwz** [comment]
Hi @aarondglover ,

Thanks for reporting this issue with Timescale.

Couple of things worth to be mentioned here.

- Feature Modifying Compressed Data is only supported on PG Version 14 and above. Please check this link. [Doc](https://docs.timescale.com/use-timescale/latest/compression/modify-compressed-data/)
- Since you are using our hosted platform, we recommend contacting support team, so that, they can gather more details like, core dumps and system logs.

**Comment by martinhale** [comment]
I'm reasonably sure that PG13 was supported for this feature in `2.11.0`? I'm currently running this combination in production and I'm fairly sure that we have successful inserts running into compressed chunks.

Maybe inserts are OK with PG13, and the updates/deletes require PG14?

Was this support removed/changed moving from `2.11.0` -> `2.11.1`, or am I mistaken?

**Comment by svenklemm** [comment]
> I'm reasonably sure that PG13 was supported for this feature in `2.11.0`? I'm currently running this combination in production and I'm fairly sure that we have successful inserts running into compressed chunks.
> 
> Maybe inserts are OK with PG13, and the updates/deletes require PG14?
> 
> Was this support removed/changed moving from `2.11.0` -> `2.11.1`, or am I mistaken?

Are your INSERTs targetting compressed chunks? There should be no functionality change in a patch release.

**Comment by aarondglover** [comment]
Sorry I did not see subsequent messages in this issue after it was suggested that version 14 was required.

@svenklemm I was attempting to perform updates on compressed data. 

I have since updated my server (TSDB MST) to version 15.4 and updated to 2.11.2 (eagerly awaiting 2.12.0 in MST)

now I have a similar error that support are trying to assist me with.  I will open a different case though. this one might be able to be closed???
